### PR TITLE
SDU-1878 smart start inclusion and node provisioning list documentation

### PR DESCRIPTION
**Pull request by eternalamenity**

### CODE REVIEW

File: thing-management.md [hunk 1]
```diff
@@ -2,7 +2,229 @@
...
+    {
+        "serv": "zwave-ad",
+        "type": "cmd.npl.qr",
+        "val_t": "str",
+        "val": "92011287329843274893247893247324",
+        "props": null,
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
```
Comment (bjhandeland) [inline thing-management.md:35]: Pref using ``` notation with language

File: thing-management.md [hunk 2]
```diff
@@ -2,7 +2,229 @@
...
+
+### Adding or editing a Z-Wave device to Node Provisioning List manually
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
+
+Message (command):
+
+    {
```
Comment (bjhandeland) [inline thing-management.md:43]: Pref ``` notation with language

File: thing-management.md [hunk 3]
```diff
@@ -2,7 +2,229 @@
...
+        "type": "cmd.npl.qr",
+        "val_t": "str",
+        "val": "92011287329843274893247893247324",
+        "props": null,
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
+
+### Adding or editing a Z-Wave device to Node Provisioning List manually
```
Comment (bjhandeland) [inline thing-management.md:37]: Can use NPL here

File: thing-management.md [hunk 4]
```diff
@@ -2,7 +2,229 @@
...
+            "name": "FH Smart Start Plug",
+            "location": "Living Room",
+            "joining_info_type": "3",
+        },
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
+
+### Removing a Z-Wave device from Node Provisioning List
```
Comment (bjhandeland) [inline thing-management.md:76]: Use NPL

File: thing-management.md [hunk 5]
```diff
@@ -2,7 +2,229 @@
...
+        "type": "cmd.npl.entry_delete",
+        "val_t": "str",
+        "val": "67656-08786-98576-65768-67656-08786-98576-65768",
+        "props": null,
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
+
+### Report for Node Provisioning List Entry
```
Comment (bjhandeland) [inline thing-management.md:93]: NPL

File: thing-management.md [hunk 6]
```diff
@@ -2,7 +2,229 @@
...
+Below report can occur in four situations:
```
Comment (bjhandeland) [inline thing-management.md:94]: Missing space between header and text

Comment (bjhandeland) [reply thing-management.md:94]: > The following report can occur in five situations:

File: thing-management.md [hunk 7]
```diff
@@ -2,7 +2,229 @@
...
+1. After adding entry (status="added").
+2. After deleting entry (status="deleted" or status="deleted_only_from_npl").
+3. After editing entry (status="edited", network_status: {"status": "not_in_network"}).
```
Comment (bjhandeland) [inline thing-management.md:97]: change to "an entry" for all three lines

File: thing-management.md [hunk 8]
```diff
@@ -2,7 +2,229 @@
...
+4. After powering up a device (status="edited", network_status: {"status": "included"}).
+5. After inclusion in wrong network (status="included_bad_network").
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
+
+Message (command):
+
+    {
```
Comment (bjhandeland) [inline thing-management.md:105]: Pref ``` with lang

File: thing-management.md [hunk 9]
```diff
@@ -2,7 +2,229 @@
...
+            "name": "FH Smart Start Plug",
+            "location": "Living Room",
+            "joining_info_type": "3",
+        },
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
+
+### Getting all entries for a Node Provisioning List
```
Comment (bjhandeland) [inline thing-management.md:141]: NPL

File: thing-management.md [hunk 10]
```diff
@@ -2,7 +2,229 @@
...
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
+
+Message (command):
+
+    {
```
Comment (bjhandeland) [inline thing-management.md:147]: use ```

File: thing-management.md [hunk 11]
```diff
@@ -2,7 +2,229 @@
...
+
+### Report after getting all entries for a Node Provisioning List
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
+
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
+
+Message (command):
+
+    {
```
Comment (bjhandeland) [inline thing-management.md:166]: use ```

**Review by bjhandeland — CHANGES_REQUESTED** [review]
Some comments

File: thing-management.md [hunk 12]
```diff
@@ -2,7 +2,229 @@
 
 The following APIs should be common across all adapters, but the example code will use zwave-ad. Notes have been added for specific differences for the most common adapters.
 
-## Adding a thing to FH system
+## Adding a thing to FH system - SmartStart (Z-Wave only)
+
+Smart Start inclusion is a process of adding devices with pre-configuration step using it's DSK printed on a back of a device.
+With this functionality, you simply mount your device and power it on.
+The devices are then automatically recognized and added to the network because they already come with a configuration set beforehand.
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md:9]: I don't understand a question, sorry

File: thing-management.md [hunk 13]
```diff
@@ -2,7 +2,229 @@
 
 The following APIs should be common across all adapters, but the example code will use zwave-ad. Notes have been added for specific differences for the most common adapters.
 
-## Adding a thing to FH system
+## Adding a thing to FH system - SmartStart (Z-Wave only)
+
+Smart Start inclusion is a process of adding devices with pre-configuration step using it's DSK printed on a back of a device.
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md [hunk 14]
```diff
@@ -2,7 +2,229 @@
...
+With this functionality, you simply mount your device and power it on.
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md [hunk 15]
```diff
@@ -2,7 +2,229 @@
...
+The devices are then automatically recognized and added to the network because they already come with a configuration set beforehand.
+
+There are two types of adding devices using SmartStart:
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md [hunk 16]
```diff
@@ -2,7 +2,229 @@
...
+1. Scanning it's DSK using mobile phone camera.
+2. Manually writing a DSK code into a Node Provisioning List of a Smarthub.
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md [hunk 17]
```diff
@@ -2,7 +2,229 @@
...
+
+What is a Node Provisioning List?
+NPL is a list which contains all the DSK's for devices that can be added using SmartStart or by a classic inclusion with Security S2 capability without additional classic inclusion security bootstraping key exchange step.
+NPL is a Z-Wave abstraction which is used mostly for SmartStart functionality but also for simplifying S2 capable nodes inclusion.
+NPL comes with a possibility to add, remove or edit it's entries.
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md (hunk 2 above)
Comment (eternalamenity) [reply thing-management.md:46]: done

File: thing-management.md (hunk 4 above)
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md (hunk 6 above)
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md (hunk 7 above)
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md (hunk 10 above)
Comment (eternalamenity) [reply thing-management.md:157]: done

File: thing-management.md [hunk 18]
```diff
@@ -2,7 +2,229 @@
...
+        "type": "cmd.npl.get_list",
+        "val_t": "null",
+        "val": null,
+        "props": null,
+        "tags": null,
+        "ctime": "1970-01-01T00:00:00+0000",
+        "uid": "123456789"
+    }
+
+### Report after getting all entries for a Node Provisioning List
```
[the remark this answers is not in the document]
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md [hunk 19]
```diff
@@ -1,8 +1,356 @@
...
+
+- `s2_only`: Device must manually be set to Learn Mode and follow S2 bootstraping instructions.
+- `smart_start`: Device will be included and S2 bootstrapped automatically using Z-Wave Smart Start.
+- `smart_start_lr`: Device will be included and S2 bootstrapped automatically using Z-Wave Long Range Smart Start.
+
+#### `network_status`
+
+Object containing nodeId and status.
+
+> `status` can be `not_in_network`, `failing` or `included`.
```
Comment (bjhandeland) [inline thing-management.md:50]: Can you add info about the different statuses here?

**Review by bjhandeland — CHANGES_REQUESTED** [review]
one more comment

File: thing-management.md (hunk 19 above)
Comment (eternalamenity) [reply thing-management.md]: done

**Pull request merged**